// @vscode/webview-ui-toolkit/dist/esm/panels/index.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { __decorate } from "tslib";
import { customElement } from '@microsoft/fast-element';
import { Tab, TabPanel, TabPanelTemplate, Tabs, TabsOrientation, TabsTemplate, TabTemplate, } from '@microsoft/fast-foundation';
import { PanelsStyles } from './panels.styles';
import { PanelTabStyles } from './panel-tab.styles';
import { PanelViewStyles } from './panel-view.styles';
/**
 * The Visual Studio Code panels component.
 *
 * @remarks
 * HTML Element: `<vscode-panels>`
 *
 * @public
 */
let VSCodePanels = class VSCodePanels extends Tabs {
    /**
     * Component lifecycle method that runs when the component is inserted
     * into the DOM.
     *
     * @internal
     */
    connectedCallback() {
        super.connectedCallback();
        // This will override any usage of the orientation attribute
        // inherited by the FAST Foundation Tabs component so that
        // VSCodePanels are always oriented horizontally
        if (this.orientation) {
            this.orientation = TabsOrientation.horizontal;
        }
        // Sets a default ARIA label on the panels component only if an
        // aria-label attribute does not already exist
        const ariaLabelValue = this.getAttribute('aria-label');
        if (!ariaLabelValue) {
            this.setAttribute('aria-label', 'Panels');
        }
    }
};
VSCodePanels = __decorate([
    customElement({
        name: 'vscode-panels',
        template: TabsTemplate,
        styles: PanelsStyles,
    })
], VSCodePanels);
export { VSCodePanels };
/**
 * The Visual Studio Code panel tab component.
 *
 * @remarks
 * HTML Element: `<vscode-panel-tab>`
 *
 * @public
 */
let VSCodePanelTab = class VSCodePanelTab extends Tab {
    /**
     * Component lifecycle method that runs when the component is inserted
     * into the DOM.
     *
     * @internal
     */
    connectedCallback() {
        super.connectedCallback();
        // This will override any usage of the disabled attribute
        // inherited by the FAST Foundation Tab component so that
        // VSCodePanelTab can never be disabled
        if (this.disabled) {
            this.disabled = false;
        }
        if (this.textContent) {
            this.setAttribute('aria-label', this.textContent);
        }
    }
};
VSCodePanelTab = __decorate([
    customElement({
        name: 'vscode-panel-tab',
        template: TabTemplate,
        styles: PanelTabStyles,
    })
], VSCodePanelTab);
export { VSCodePanelTab };
/**
 * The Visual Studio Code panel view component.
 *
 * @remarks
 * HTML Element: `<vscode-panel-view>`
 *
 * @public
 */
let VSCodePanelView = class VSCodePanelView extends TabPanel {
};
VSCodePanelView = __decorate([
    customElement({
        name: 'vscode-panel-view',
        template: TabPanelTemplate,
        styles: PanelViewStyles,
    })
], VSCodePanelView);
export { VSCodePanelView };
