// @vscode/webview-ui-toolkit/dist/esm/design-tokens.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { DesignToken } from '@microsoft/fast-foundation';
const { create } = DesignToken;
/**
 * Global design tokens.
 */
export const designUnit = create('design-unit').withDefault(4);
export const borderWidth = create('border-width').withDefault(1);
export const contrastActiveBorder = create('contrast-active-border').withDefault('#f38518');
export const contrastBorder = create('contrast-border').withDefault('#6fc3df');
export const cornerRadius = create('corner-radius').withDefault(0);
export const disabledOpacity = create('disabled-opacity').withDefault(0.4);
export const focusBorder = create('focus-border').withDefault('#007fd4');
export const foreground = create('foreground').withDefault('#cccccc');
export const fontFamily = create('font-family').withDefault('-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol');
export const fontWeight = create('font-weight').withDefault('400');
export const inputHeight = create('input-height').withDefault('26');
export const inputMinWidth = create('input-min-width').withDefault('100px');
/**
 * Type-ramp font-size and line-height design tokens.
 */
export const typeRampBaseFontSize = create('type-ramp-base-font-size').withDefault('13px');
export const typeRampBaseLineHeight = create('type-ramp-base-line-height').withDefault('normal');
export const typeRampMinus1FontSize = create('type-ramp-minus1-font-size').withDefault('11px');
export const typeRampMinus1LineHeight = create('type-ramp-minus1-line-height').withDefault('16px');
export const typeRampMinus2FontSize = create('type-ramp-minus2-font-size').withDefault('9px');
export const typeRampMinus2LineHeight = create('type-ramp-minus2-line-height').withDefault('16px');
export const typeRampPlus1FontSize = create('type-ramp-plus1-font-size').withDefault('16px');
export const typeRampPlus1LineHeight = create('type-ramp-plus1-line-height').withDefault('24px');
/**
 * Badge design tokens.
 */
export const badgeForeground = create('badge-foreground').withDefault('#ffffff');
export const badgeBackground = create('badge-background').withDefault('#4d4d4d');
/**
 * Button design tokens.
 */
export const buttonPrimaryForeground = create('button-primary-foreground').withDefault('#ffffff');
export const buttonPrimaryBackground = create('button-primary-background').withDefault('#0e639c');
export const buttonPrimaryHoverBackground = create('button-primary-hover-background').withDefault('#1177bb');
export const buttonSecondaryForeground = create('button-secondary-foreground').withDefault('#ffffff');
export const buttonSecondaryBackground = create('button-secondary-background').withDefault('#3a3d41');
export const buttonSecondaryHoverBackground = create('button-secondary-hover-background').withDefault('#45494e');
export const buttonPaddingHorizontal = create('button-padding-horizontal').withDefault('11px');
export const buttonPaddingVertical = create('button-padding-vertical').withDefault('6px');
export const buttonIconBackground = create('button-icon-background').withDefault('transparent');
export const buttonIconHoverBackground = create('button-icon-hover-background').withDefault('rgba(90, 93, 94, 0.31)');
export const buttonIconPadding = create('button-icon-padding').withDefault('3px');
export const buttonIconCornerRadius = create('button-icon-corner-radius').withDefault('5px');
export const buttonIconFocusBorderOffset = create('button-icon-outline-offset').withDefault(0);
// Note: Button Border is used only for high contrast themes and should be
// left as transparent otherwise.
export const buttonBorder = create('button-border').withDefault('transparent');
/**
 * Checkbox design tokens.
 */
export const checkboxBackground = create('checkbox-background').withDefault('#3c3c3c');
export const checkboxForeground = create('checkbox-foreground').withDefault('#f0f0f0');
export const checkboxBorder = create('checkbox-border').withDefault('#3c3c3c');
export const checkboxCornerRadius = create('checkbox-corner-radius').withDefault(3);
/**
 * Data Grid design tokens
 */
export const listActiveSelectionBackground = create('list-active-selection-background').withDefault('#094771');
export const listActiveSelectionForeground = create('list-active-selection-foreground').withDefault('#ffffff');
export const listHoverBackground = create('list-hover-background').withDefault('#2a2d2e');
export const quickInputBackground = create('quick-input-background').withDefault('#252526');
/**
 * Divider design tokens.
 */
export const dividerBackground = create('divider-background').withDefault('#454545');
/**
 * Dropdown design tokens.
 */
export const dropdownBackground = create('dropdown-background').withDefault('#3c3c3c');
export const dropdownForeground = create('dropdown-foreground').withDefault('#f0f0f0');
export const dropdownBorder = create('dropdown-border').withDefault('#3c3c3c');
export const dropdownListMaxHeight = create('dropdown-list-max-height').withDefault('200px');
/**
 * Text Field & Area design tokens.
 */
export const inputBackground = create('input-background').withDefault('#3c3c3c');
export const inputForeground = create('input-foreground').withDefault('#cccccc');
export const inputPlaceholderForeground = create('input-placeholder-foreground').withDefault('#cccccc');
/**
 * Link design tokens.
 */
export const linkForeground = create('link-active-foreground').withDefault('#3794ff');
export const linkActiveForeground = create('link-foreground').withDefault('#3794ff');
/**
 * Progress Bar & Ring design tokens.
 */
export const progressBackground = create('progress-background').withDefault('#0e70c0');
/**
 * Panels design tokens.
 */
export const panelViewBackground = create('panel-view-background').withDefault('#1e1e1e');
export const panelViewBorder = create('panel-view-border').withDefault('#80808059');
export const panelTabForeground = create('panel-tab-foreground').withDefault('#e7e7e799');
export const panelTabActiveForeground = create('panel-tab-active-foreground').withDefault('#e7e7e7');
export const panelTabActiveBorder = create('panel-tab-active-border').withDefault('#e7e7e7');
/**
 * Tag design tokens.
 */
export const tagCornerRadius = create('tag-corner-radius').withDefault('2px');
